src/ripple/app/ledger/InboundLedger.cpp

```diff
@@ -161,14 +161,15 @@ bool InboundLedger::tryLocal ()
 
             if (m_journal.trace) m_journal.trace <<
                 "Ledger header found in fetch pack";
-            mLedger = std::make_shared<Ledger> (data, true);
+            mLedger = std::make_shared<Ledger> (
+                data.data(), data.size(), true);
             getApp().getNodeStore ().store (
                 hotLEDGER, std::move (data), mHash);
         }
         else
         {
-            mLedger = std::make_shared<Ledger> (
-                strCopy (node->getData ()), true);
+            mLedger = std::make_shared<Ledger>(
+                node->getData().data(), node->getData().size(), true);
```

Comment:
What about something like `make_shared<Ledger>(toSlice(node->getData())`? Change would stay with ledger (except for the `toShared` helper classes.